packages/babel-parser/src/tokenizer/index.js

```diff
@@ -1197,7 +1197,7 @@ export default class Tokenizer extends ParserErrors {
       const code = this.input.charCodeAt(this.state.pos);
       let val;
 
-      if (code === charCodes.underscore) {
+      if (code === charCodes.underscore && allowNumSeparator !== "bail") {
```

Comment:
The invalid JSX syntax `&#0_0;` is recoverable: we can either pass `allowNumSeparator: false` and refine the error message of `NumericSeparatorInEscapeSequence`, or pass a context string so we can throw different errors. Though personally I think html entities are just like escape sequences in JS.

Reply:
It's not invalid, it's rendeded as-is and not as an HTML entity.